### DISCUSSION TITLE
Upload file gives body error with UploadFile

### DISCUSSION THREAD

**Discussion by llorencs** [fastapi/fastapi#7062] [Questions] [answered]
### Description
I am having issues uploading a file to a FastAPI server. I found similar issues:
#1367  and #405

However, there was no solution. The issue is that it works with small sized files, but when a file is bigger than 100Kb it does not work. It looks like the reading of the file does not finish before trying to run the other tasks like writing.

### Code sample
```
import uvicorn
from fastapi import FastAPI, File, UploadFile
from fastapi.responses import FileResponse
import aiofiles

app = FastAPI(title='Untitled', description='doThings', version='0.0.1')

uploads = 'uploads'
uploads_dir = pathlib.Path(os.getcwd(), uploads)

@app.post('/api/upload/')
async def upload_file(file: UploadFile=File(...)):
    """
    Upload the file to be processed.
    """
    print(file)
#    async with aiofiles.open(file.file, 'rb') as fin:
#       exfile = fin.read() 
    file_name = pathlib.Path(uploads_dir, file.filename)
    async with aiofiles.open(f'{file_name}', 'wb') as f:
        await f.write(await file.read())
    return {'filename': file.filename, 
                    'content-type': file.content_type}

if __name__ == '__main__':
    uvicorn.run('main:app', host='127.0.0.1', port=5003, log_level='info')
```
### Output

The error happens just when it tries to read the file. I have tried it only with the return and it does not work, neither. I have run many tests and I don't get where the error comes.

This is the output of the uvicorn server:

```
[32mINFO[0m:     Started server process [[36m7108[0m]
[32mINFO[0m:     Waiting for application startup.
[32mINFO[0m:     Application startup complete.
[32mINFO[0m:     Uvicorn running on [1mhttp://127.0.0.1:5003[0m (Press CTRL+C to quit)
[32mINFO[0m:     127.0.0.1:52854 - "[1mPOST /api/upload HTTP/1.1[0m" [33m307 Temporary Redirect[0m
Did not find boundary character 80 at index 2
[32mINFO[0m:     127.0.0.1:52855 - "[1mPOST /api/upload/ HTTP/1.1[0m" [31m400 Bad Request[0m
[33mWARNING[0m:  Invalid HTTP request received.
```

And in Postman I get this error message:

```
{
    "detail": "There was an error parsing the body"
}
```

### Environment

OS: Windows 10
Python: Python 3.8.6
FastApi 0.61.2
Starlette 0.13.6
Pydantic 1.7.2

**Comment by Mause** [comment]
What are you using to upload the file?

**Comment by llorencs** [comment]
I made the tests with Postman v7.36.0. 
![imagen](https://user-images.githubusercontent.com/6440040/100193934-163e9f00-2ef5-11eb-99ed-cae94f281928.png)

**Comment by chrisdietr** [comment]
@llorencs I think it's a postman bug.

I've adjusted your script a bit (added `os` and `pathlib``modules as well as manually created the `uploads` folder).

![2020-11-29 at 22 54-2](https://user-images.githubusercontent.com/2683346/100553593-5efcac00-3297-11eb-87fb-6b412a006da0.png)

Then tested with small file in postman: worked.
Then tested big file (16mb) in postman: fail
Then tested big file (16mb) with https://insomnia.rest/: worked

![2020-11-29 at 23 09](https://user-images.githubusercontent.com/2683346/100553695-e77b4c80-3297-11eb-906c-53b916004916.png)


Postman bug documented here and seems to be still valid: https://github.com/postmanlabs/postman-app-support/issues/576

**Comment by llorencs** [comment]
Hello, following this, I have tested it directely with a requests snippet and it worked correctly. So, it can be closed.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:

**Comment by boun** [comment]
Sorry for commenting something totally unrelated to a closed bug, but google brought me here. 

I had the same error, but did not use postman. In my case it was a kubernetes deployment. The solution was that the helm chart set the **root fs to readonly**. 

Hope someone finds this helpful in the future.

**Comment by johnPractice** [comment]
you can use the "insomnia" app instead of Postman
I have the same problem with Postman added some headers automatically and broke my API services

**Comment by IDmedia** [comment, marked as answer]
I know this is old, but I encountered the same issue and noticed I used `/upload/` in FastAPI and `/upload` in Postman.
Once I removed the trailing slash in my code it worked from Postman as well. It didn't work to append the slash in Postman for some reason.

**Reply by lucas-labs** [reply]
You just saved me hours (or days).  Thanks!

**Reply by Sidhant-raj** [reply]
Really you saved me hours of work. Thanks a lot

**Reply by ArtisStella** [reply]
This issue is apparently not related to postman but FastAPI itself.

After trying to send a file from an ExpressJS API to FastAPI to no success for hours, I decided to use postman which also didn't work.
Then I tried creating an identical endpoint in my ExpressJS API, and an identical request in postman, it worked. 
I almost gave up on FastAPI until I came across this simple fix, added the / to both ExpressJS API and Postman and both worked.

**Reply by hanieas** [reply]
I really appreciate your answer,  the issue is resolved by adding `/` to the end of the endpoint in the Postman.

**Reply by juanitoddd** [reply]
Thanks, I cant believe this was the issue

**Reply by jacobgawel** [reply]
yup same here, this fixed the issue for me

**Comment by shashstormer** [comment]
basic usage of upload file has been used in this project https://github.com/shashstormer/aiohttp-file-helper/blob/master/__fast_api__.py
you can check out the implementation
